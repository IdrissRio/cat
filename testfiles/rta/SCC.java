// PackageName: org.extendj.callgraph.SCC
// MethodName: main

package org.extendj.callgraph;
public class SCC {
  public static void main(String[] args) {
    Root r = new Root();
    r.a1();
  }
}

class Root {
  public void a1() { a2(); }
  public void a2() {
    a1();
    a3();
  }
  public void a3() { a4(); }
  public void a4() { a3(); }
}
