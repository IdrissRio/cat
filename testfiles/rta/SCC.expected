java.lang.Object::<init>()
org.extendj.callgraph.Root::Root() -> java.lang.Object::<init>()
org.extendj.callgraph.Root::a1() -> org.extendj.callgraph.Root::a2()
org.extendj.callgraph.Root::a2() -> org.extendj.callgraph.Root::a1(), org.extendj.callgraph.Root::a3()
org.extendj.callgraph.Root::a3() -> org.extendj.callgraph.Root::a4()
org.extendj.callgraph.Root::a4() -> org.extendj.callgraph.Root::a3()
org.extendj.callgraph.SCC::main(String[]) -> org.extendj.callgraph.Root::Root(), org.extendj.callgraph.Root::a1()

